src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> currentNode = table[getKeyIndex(key)];
+        if (currentNode == null) {
+            table[getKeyIndex(key)] = new Node<>(key, value, null);
```

Comment:
you are calling method getKeyIndex() twice. You can save the value in the variable and reuse it. Method getKeyIndex can be renamed to getIndex() as you searching index not just for the key but for the whole node.

Reply:
You can add return here and not use else

Reply:
done